File: Domain/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Order
    {
        public Guid OrderId { get; set; }

        public string OrderNumber { get; set; }
        public DateTime Orderdate { get; set; }
        public double OrderTotal { get; set; }
        public string OrderStatus { get; set; }
        public string OrderComment { get; set; }
        public Customer Customer { get; set; }

        public ICollection<OrderItem> OrderItems { get; set; }
    }
}

File: client-app/src/app/models/order.ts
import { Customer } from "./customer";
import { OrderedProduct } from "./orderedProduct";

export interface Order {
    orderId: string,
    orderNumber: string,
    orderDate: Date,
    orderTotal: number,
    orderStatus: string,
    orderComment?: string,
    customer: Customer,
    orderedProducts: OrderedProduct[]
  }
  

  export interface OrderToUpdate {
    orderId: string,
    orderNumber: string,
    orderDate: Date,
    orderStatus: string,
    orderComment?: string,
    orderTotal: number
  }